functions/src/utils/storage.ts

```diff
@@ -0,0 +1,31 @@
+import {app} from '../app';
+import {v4 as uuidv4} from 'uuid';
+
+/**
+ * Uploads a base64 encoded image to Google Cloud Storage.
+ * @param base64Data The base64 encoded image string.
+ * @param mimeType The MIME type of the image.
+ * @param prefix A prefix for the file name in the bucket.
+ * @returns The public URL of the uploaded image.
+ */
+export async function uploadBase64ImageToGCS(
+  base64Data: string,
+  mimeType: string,
+  prefix = 'generated-images',
+): Promise<string> {
+  const BUCKET_NAME = 'msgraham-deliberate-labs';
```

Comment:
Consider making this more generic, like `deliberate-lab` or `pair-deliberate-lab`?